### ISSUE TITLE
Remove deprecated code after the 1.6 release

### ISSUE THREAD

**Issue by akihironitta** [Lightning-AI/lightning#12521]
## Proposed refactor

Remove deprecated code after the 1.6 release.

**NOTE**: Please pick up a _single_ item from the list (by commenting here in the issue) - and if there are no conflicts - we will happily assign you and put your name in front of the item in the list.

Please note that unless mentioned, the classes are importable from `pytorch_lightning`, example: `from pytorch_lightning import Trainer`.

* [x] `LightningModule.summarize` -> #12559
* [x] `pytorch_lightning.core.memory.LayerSummary` -> #12593
* [x] `pytorch_lightning.core.memory.ModelSummary` -> #12593
* [x] `pytorch_lightning.core.memory.get_gpu_memory_map` -> #12644
* [x] `pytorch_lightning.core.memory.get_memory_profile` -> #12659
* [x] `LightningModule.model_size` -> #12641
* [x] `LightningDataModule.train_transforms` -> #12662
* [x] `LightningDataModule.val_transforms` -> #12763
* [x] `LightningDataModule.test_transforms` -> #12773
* [x] `LightningDataModule.size` -> #12780
* [x] `LightningDataModule.dims` and `LightningDataModule(dims=...)` -> #12780
* [x] `LightningModule.get_progress_bar_dict` -> #12839 
* [x] `Trainer.progress_bar_dict` -> #12839 
* [x] `Trainer(prepare_data_per_node=...)` -> #12536
* [x] `Trainer(stochastic_weight_avg=...)` -> #12535
* [x] `Trainer(terminate_on_nan=...)` and `Trainer.terminate_on_nan` -> #12553 
* [x] `LightningModule.on_{train,val,test,predict}_dataloader` -> #13033
* [x] `pytorch_lightning.loggers.TestTubeLogger` -> #12859
* [x] `pytorch_lightning.Callback.on_keyboard_interrupt` -> #13438
* [x] `Trainer(process_position=...)` -> #13071
* [x] `Trainer(flush_logs_every_n_steps=...)` -> #13074
* [x] `LightningModule.add_to_queue` -> @shenoynikhil
* [x] `LightningModule.get_from_queue` -> @shenoynikhil
* [x] `Trainer(progress_bar_refresh_rate=...)` -> #12514
* [x] `LightningLoggerBase.close` and `pytorch_lightning.loggers.LoggerCollection.close` -> #13149
* [x] `pytorch_lightning.distributed.dist.LightningDistributed` #13549
* [x] `Trainer(checkpoint_callback=...)` -> #13027
* [x]  Passing `dataloader_idx` to `on_train_batch_start` of `pytorch_lightning.Callback` and `LightningModule`  -> #12769
* [x] `LightningModule.on_post_move_to_device` #13548
* [x] `pytorch_lightning.core.decorators.parameter_validation` #13514
* [x] `Trainer(accelerator="ddp_spawn")` #12696
* [x] `Trainer(plugins="ddp_spawn")` #12700
* [x] `Trainer(weights_summary="full")`, `Trainer(weights_summary=None)`, `Trainer.weights_summary` -> #13070
* [x] `Trainer(log_gpu_memory=...)` -> #12657
* [x] `Trainer.slurm_job_id` #13459
* [x] `pytorch_lightning.callbacks.gpu_stats.GPUStatsMonitor` -> #12554
* [x] `pytorch_lightning.callbacks.gpu_stats.XLAStatsMonitor` -> #12688
* [x] `pytorch_lightning.callbacks.progress.ProgressBar` -> #12658
* [x] `Trainer(max_steps=None)` and `Trainer.fit_loop.max_steps = None` #13591
* [x] `pytorch_lightning.callbacks.lr_monitor.LearningRateMonitor.lr_sch_names` -> #13353
* [x] `KubeflowEnvironment.is_using_kubeflow, LSFEnvironment.is_using_lsf, TorchElasticEnvironment.is_using_torchelastic` #13458
* [x] `pytorch_lightning.overrides.distributed.IndexBatchSamplerWrapper.batch_indices` #13565
* [x] `pytorch_lightning.strategies.SingleDeviceStrategy.post_dispatch` #13461
* [x] `pytorch_lightning.trainer.connectors.logger_connector.logger_connector.LoggerConnector.gpu_metrics`

Feel free to cross-check from the [test file](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/deprecated_api/test_remove_1-7.py) to ensure that the relevant test fails now (since it's no more deprecated and instead removed).

### Pitch
All the deprecated features we have are tested here:

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/deprecated_api/test_remove_1-7.py

If you are interested in taking care of one item, post a comment here asking to take it. This avoids multiple people working on the same thing.

### Additional context
See pull requests linked in #10312 for examples on how to contribute :)
Or a recent pull request #12514.

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @justusschock @awaelchli @rohitgr7 @krshrimali

**Comment by krshrimali** [comment]
@akihironitta - Thanks for creating this issue. Let's create a tracker here, to help track better? What do you think?

**Comment by Borda** [comment]
> Thanks for creating this issue. Let's create a tracker here, to help track better? What do you think?

exactly, let's include in the issue list aka todo of heat methods/attributes shall be removed so individual contributors can pick one and we reduce collision of people working on the same... :rabbit:

**Comment by krshrimali** [comment]
> 

@Borda - Done! 🚀 Note and a request to maintainers: please cross-check this list.

**Comment by krshrimali** [comment]
> @krshrimali thanks for the suggestion, I have edited the comment to only include one flag.

Assigned to you. Thank you for understanding :) Please let us know if you have any queries, you can also join our slack if you wish: https://join.slack.com/t/pytorch-lightning/shared_invite/zt-12iz3cds1-uyyyBYJLiaL2bqVmMN7n~A. :)

**Comment by aniketmaurya** [comment]
Hi, I would like to take `LightningModule.summarize`

**Comment by karthikrangasai** [comment]
Hello, 
I can start with `Trainer(terminate_on_nan=...)` and `Trainer.terminate_on_nan`

**Comment by kaushikb11** [comment]
Note to the contributors here:

It's a good practice to check if any open source projects are using the API, you are planning to remove on grep.app. So, that we could notify them to use an alternative or update the code. :) 

For instance: https://grep.app/search?q=GPUStatsMonitor

**Comment by aniketmaurya** [comment]
Please assign me `pytorch_lightning.core.memory.LayerSummary` and `ModelSummary`

**Comment by puhuk** [comment]
Can I take `pytorch_lightning.core.memory.get_gpu_memory_map`

**Comment by rohitgr7** [comment]
sure, go ahead @puhuk !!!

**Comment by rschireman** [comment]
I'll take `pytorch_lightning.callbacks.progress.ProgressBar` if that's ok

**Comment by tkonopka** [comment]
Hello, I can work on `LightningModule.model_size` (?)

**Comment by krshrimali** [comment]
@ankitaS11 @rschireman @tkonopka - Thank you for showing interest! 🚀 ⚡ Please go ahead, I've assigned the respective tasks to you all (updated the description). Also, make sure that you join [our slack channel](https://join.slack.com/t/pytorch-lightning/shared_invite/zt-12iz3cds1-uyyyBYJLiaL2bqVmMN7n~A) - feel free to ask questions there if you have any!

**Comment by myxik** [comment]
Hey! I would like to take `LightningDataModule.train_transforms` as well as `val_transforms` and `test_transforms`.

**Comment by akihironitta** [comment]
Hey @myxik! Thank you for your interest! That'll be fantastic! Assigned `LightningDataModule.train_transforms` to you as your first task :)

**Comment by rohitgr7** [comment]
just an FYI here:

whenever anyone creates a PR, make sure to add
```
Part of #12521
```
instead of
```
Fixes #12521
```
in the description.

Else this issue will be linked to the PR and will be closed once it gets merged.

**Comment by vumichien** [comment]
I'll take `LightningDataModule.dims` and `LightningDataModule(dims=...)`

**Comment by rschireman** [comment]
I'll follow up #12763 by taking `LightningDataModule.test_transforms`

**Comment by vumichien** [comment]
Hi @akihironitta, I think I have to take  `LightningDataModule.size` as well because `LightningDataModule.size` will call dim variable. How do you think about it?
https://github.com/PyTorchLightning/pytorch-lightning/blob/ae3226ced96e2bc7e62f298d532aaf2290e6ef34/pytorch_lightning/core/datamodule.py#L128-L139

**Comment by rschireman** [comment]
I'll keep going down the list, I'll start working on `LightningDataModule.size`

**Comment by vumichien** [comment]
Hi @rschireman, I have made PR with `LightningDataModule.size` and `LightningDataModule.dims`, so please work with another one. If you work with `LightningDataModule.size`, I have to revert my work. Thank you for your understanding.

**Comment by rschireman** [comment]
@vumichien sounds great, I'll work on something else, sorry for the confusion!

**Comment by rschireman** [comment]
working on `LightningModule.get_progress_bar_dict`

**Comment by akihironitta** [comment]
> Hi @akihironitta, I think I have to take `LightningDataModule.size` as well because `LightningDataModule.size` will call dim variable. How do you think about it?

Hi @vumichien! Sounds good to me! Thanks for having done so already in your PR #12780!

**Comment by rschireman** [comment]
In `pytorch_lightning/callbacks/progress/base.py` I see that the `get_progress_bar_dict()` function is called, I'm having some trouble on what should it be replaced with so the `get_metrics()` function doesn't produce an error :sweat_smile: 


```
  def get_metrics(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> Dict[str, Union[int, str]]:
        r"""
        Combines progress bar metrics collected from the trainer with standard metrics from get_standard_metrics.
        Implement this to override the items displayed in the progress bar.
        Here is an example of how to override the defaults:
        .. code-block:: python
            def get_metrics(self, trainer, model):
                # don't show the version number
                items = super().get_metrics(trainer, model)
                items.pop("v_num", None)
                return items
        Return:
            Dictionary with the items to be displayed in the progress bar.
        """
        standard_metrics = pl_module.get_progress_bar_dict()
        pbar_metrics = trainer.progress_bar_metrics
        duplicates = list(standard_metrics.keys() & pbar_metrics.keys())
        if duplicates:
            rank_zero_warn(
                f"The progress bar already tracks a metric with the name(s) '{', '.join(duplicates)}' and"
                f" `self.log('{duplicates[0]}', ..., prog_bar=True)` will overwrite this value. "
                " If this is undesired, change the name or override `get_metrics()` in the progress bar callback.",
            )

        return {**standard_metrics, **pbar_metrics}
```

**Comment by rohitgr7** [comment]
@rschireman 
```py
standard_metrics = pl_module.get_progress_bar_dict()
```
should be:
```py
standard_metrics = get_standard_metrics(trainer, pl_module)
```

**Comment by rschireman** [comment]
@rohitgr7 
Thank you!

**Comment by stanbiryukov** [comment]
I can take removing deprecated `pytorch_lightning.loggers.TestTubeLogger`

**Comment by rschireman** [comment]
Let's keep the ball rolling, I'll take `LightningModule.on_train_dataloader`

**Comment by kingjuno** [comment]
@akihironitta Can i work on `LightningModule.on_val_dataloader` or `LightningModule.on_test_dataloader`

**Comment by shenoynikhil98** [comment]
@akihironitta Can I work on `Trainer(checkpoint_callback=...)`?

**Comment by akihironitta** [comment]
@kingjuno @shenoynikhil98 Thank you! That'll be great :)

**Comment by shenoynikhil98** [comment]
@akihironitta Can you also assign `Trainer(weights_summary="full"), Trainer(weights_summary=None), Trainer.weights_summary` task to me?

**Comment by rohitgr7** [comment]
@shenoynikhil users should pass in `ModelSummary(max_depth=...)`. You can improve the `max_depth` docstrings to mention more details about it while removing `weights_summary`.

**Comment by shenoynikhil** [comment]
@akihironitta Can you please assign `Trainer(process_position=...)` to me.

**Comment by shenoynikhil** [comment]
@akihironitta #12688 Removes XLAMonitor, can you please strike `pytorch_lightning.callbacks.gpu_stats.XLAStatsMonitor` off

**Comment by madaanpulkit** [comment]
@akihironitta Can you please assign `Trainer(max_steps=None)` and `Trainer.fit_loop.max_steps = None` to me.

**Comment by shenoynikhil** [comment]
@akihironitta Can you please assign `Trainer(flush_logs_every_n_steps=...)` to me?

**Comment by ti-ginkgo** [comment]
@akihironitta Can you please assign `LightningLoggerBase.close` to me ?

**Comment by shenoynikhil** [comment]
@akihironitta Can you assign `LightningModule.add_to_queue` and `LightningModule.get_from_queue` to me?

**Comment by nitinramvelraj** [comment]
I can take  pytorch_lightning.Callback.on_keyboard_interrupt

**Comment by rohitgr7** [comment]
yep @rschireman go ahead. I see a few more assigned to you. Are you working on them too?

**Comment by rohitgr7** [comment]
oh, I didn't see it since it wasn't marked. Go ahead with the next one :)

**Comment by akihironitta** [comment]
@shenoynikhil @nitinramvelraj Thanks for showing your interest in contributing! Let me know if you need help with the ones assigned to you :]

**Comment by akihironitta** [comment]
Hi @nitinramvelraj Thanks for asking! Let's discuss your issue via PyTorch Lightning Slack! My username there is the same as my GitHub username @akihironitta. Please DM me on Slack :) (You can join our Slack via https://pytorchlightning.ai/community if you haven't)

**Comment by akihironitta** [comment]
Hi @nitinramvelraj! Sorry for the delay. Could you try this link?
https://join.slack.com/t/pytorch-lightning/shared_invite/zt-1b6lkirxt-9b3ehI0P_31YIqQ4_Dum1A

(The link in the https://pytorchlightning.ai/community will be updated to this new link shortly.)

**Comment by shantam-8** [comment]
Hello @akihironitta, can I please take `pytorch_lightning.core.decorators.parameter_validation`?

**Comment by rohitgr7** [comment]
hey @shantam-8 ! please go ahead with a PR !!

**Comment by akihironitta** [comment]
Hi @shenoynikhil @madaanpulkit! Thank you for your interest in contributing! Let me know if you need help with the ones assigned to you :)

**Comment by shenoynikhil** [comment]
@akihironitta Have been busy for sometime, will work on it today :).

**Comment by samz5320** [comment]
Hi ! can i take `pytorch_lightning.overrides.distributed.IndexBatchSamplerWrapper.batch_indices` ? if its not assigned  .

**Comment by akihironitta** [comment]
Hi @samz5320, thank you for showing your interest! Please go ahead :)

**Comment by carmocca** [comment]
Closing. All the items are taken (some yet to be merged). Thank you all for your help!
